WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentReceiptEmailParameterDeterminer.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import Yosemite
+
+/// Determines the email to be set (if any) on a payment receipt depending on the current payment plugins (WCPay, Stripe) configuration
+/// 
+struct PaymentReceiptEmailParameterDeterminer {
+    private let paymentsPluginsDataProvider: PaymentsPluginsDataProviderProtocol
+    private let stores: StoresManager
+
+    init(paymentsPluginsDataProvider: PaymentsPluginsDataProviderProtocol = PaymentsPluginsDataProvider(),
+         stores: StoresManager = ServiceLocator.stores) {
+        self.paymentsPluginsDataProvider = paymentsPluginsDataProvider
+        self.stores = stores
+    }
+
+    /// We do not need to set the receipt email if WCPay is installed and active
+    /// and its version is higher or equal than 4.0.0, as it does it itself in that case.
+    ///
+    /// - Parameters:
+    ///   - order: the order associated with the payment
+    ///   - onCompletion: closure invoked with the result of the inquiry, containg the email (if any) or error
+    ///
+    func receiptEmail(from order: Order, onCompletion: @escaping ((Result<String?, Error>) -> Void)) {
+        synchronizePlugins(from: order.siteID) { result in
```

Comment:
Thanks for the pointing that out @jaclync :) We also thought about it [here](https://github.com/woocommerce/woocommerce-ios/pull/6585#discussion_r846008429)

>  it does ensure that the plugins are up-to-date, but if it fails it might result in an error state for the whole payment flow.

Actually, if there is an error syncing we do not stop the whole payment flow, we just returned nil for email in `PaymentCaptureOrchestrator`:

```
paymentReceiptEmailParameterDeterminer.receiptEmail(from: order) { result in
            var receiptEmail: String?
            if case let .success(email) = result {
                receiptEmail = email
            }
```
About syncing, it is true that it creates an extra request, but I wanted the class to be totally independent and avoid any assumptions about previous syncing (e.g if someone removes the syncing at launch this would break without us knowing it)
Furthermore, users can change the plugins configuration while the app is running, an unlikely case but possible anyways. I tried the request with bad network conditions, and still, it doesn't take very long.
As you said, we can always remove it if we detect that this is suboptimal, e.g when adding the onboarding checks to the payment collection in orders. I will add a note in that issue 🚀

Reply:
Sounds great, thanks for the detailed explanation!